util.go

```diff
@@ -23,6 +23,27 @@ func isFalse(value interface{}) bool {
 	case nil:
 		return true
 	}
+	// Try the reflection cases before returning false.
+	v := reflect.ValueOf(value)
+	vType := v.Type().Kind()
+	if vType == reflect.Struct {
```

Comment:
Looks good, if you wanted to simplify this you could use to a switch statement, and drop the usage of the vType i think. This also adds support for a map[X]Y with 0 len. String doesn't strictly need to be the map key.

```go
	// Try the reflection cases before returning false.
	rv := reflect.ValueOf(value)
	switch rv.Kind() {
	case reflect.Struct:
		// A struct type will never be false, even if
		// all of its values are the zero type.
		return false
	case reflect.Slice, reflect.Map:
		return rv.Len() == 0
	case reflect.Ptr:
		if rv.IsNil() {
			return true
		}
		// If it's a pointer type, we'll try to deref the pointer
		// and evaluate the pointer value for isFalse.
		element := rv.Elem()
		return isFalse(element.Interface())
	}
```